src/core/control/shaperecognizer/ShapeRecognizer.cpp

```diff
@@ -242,13 +244,43 @@ void ShapeRecognizer::optimizePolygonal(const Point* pt, int nsides, int* breaks
     }
 }
 
+auto ShapeRecognizer::isStrokeLargeEnough(Stroke* stroke, double strokeMinSize) -> bool {
+    if (stroke->getPointCount() < 3) {
+        return false;
+    }
+
+    double Xmax = std::numeric_limits<double>::min();
+    double Xmin = std::numeric_limits<double>::max();
+
+    double Ymax = std::numeric_limits<double>::min();
+    double Ymin = std::numeric_limits<double>::max();
+
+    for (auto& point: stroke->getPointVector()) {
+        if (point.x > Xmax)
+            Xmax = point.x;
+        if (point.x < Xmin)
+            Xmin = point.x;
+
+        if (point.y > Ymax)
+            Ymax = point.y;
+        if (point.y < Ymin)
+            Ymin = point.y;
+    }
```

Comment:
You can retrieve this size by calling `stroke->getSnappedBounds()` The calculation is the same (thus removing code duplication), and the result is cached for later use (in Element::snappedBounds).

Reply:
:+1: